### PR TITLE
INFRA-111: GitHub Actions Runner Windows images with Packer

### PR DESCRIPTION
**Pull request by BenedekFarkas**

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -10,15 +10,15 @@ on:
 jobs:
   call-build-and-test-workflow:
     name: Build and Test - root solution
+    runs-on: TC-mod
```
Comment (Piedone) [inline .github/workflows/build-and-test.yml]: `machine-types` is for the same thing.

Comment (BenedekFarkas) [reply .github/workflows/build-and-test.yml]: Is it deprecated? It's not even in the documentation. I Googled
`github actions workflow "machine-types"`
and funny enough, the second result is this: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/2763199416/workflow
It says:
`The workflow is not valid. .github/workflows/build-and-test.yml (Line: 22, Col: 9): Unexpected value 'with' .github/workflows/build-and-test.yml (Line: 19, Col: 5): Required property is missing: runs-on`

EDIT: Never mind, it's our own custom parameter. Still confusing a bit, naming it completely different that what's used downstream.

Comment (Piedone) [reply .github/workflows/build-and-test.yml]: It's named differently because it works differently ("machine type" is how GH also calls these values).

**Comment by Piedone** [comment]
On the next commit, UI testing will break if you don't merge from `dev` beforehand.

**Pull request merged**